chain-tx-enclave-next/enclave-ra/ra-sp-client/src/client.rs

```diff
@@ -76,7 +77,20 @@ impl SpRaClient {
             .ok_or(SpRaClientError::NoResponse)?;
 
         match response {
-            Response::GetAttestationReport { attestation_report } => Ok(attestation_report),
+            Response::GetAttestationReport { attestation_report } => {
+                // validate nonce
+                let body: AttestationReportBody = serde_json::from_slice(&attestation_report.body)?;
+                let ret_nonce = body.nonce.ok_or_else(|| {
+                    SpRaClientError::Nonce("No nonce found in attestation report.".to_string())
+                })?;
+                if ret_nonce.as_bytes() == nonce {
```

Comment:
probably ok (not sure if this timing leak can be hijacked in enclaves), but perhaps constant time comparison for sanity.

Reply:
Putting all my faith in sgx :smile:. Besides, won't LLVM muck up things anyway, if we do a constant time comparison?

Reply:
`subtle` tries to prevent the compiler from optimizing away constant time comparisons: https://doc.dalek.rs/src/subtle/lib.rs.html#211 so with that lib or similar, it shouldn't